test/functional/terminal/channel_spec.lua

```diff
@@ -59,6 +59,10 @@ describe('terminal channel is closed and later released if', function()
   end)
 
   it('opened by termopen(), exited, and deleted by pressing a key', function()
+    if helpers.isCI('sourcehut') then
+      pending('Flaky test on freebsd.', function() end)
+    end
```

Comment:
which part of this is failing? maybe a retry() makes sense instead, for one of the eq().

We need to ask similiar questions for all cases here.

Reply:
> which part of this is failing?
 
I didn't check. I'll start over and note which tests are failing. 

> maybe a retry() makes sense instead, for one of the eq().

Sure, but that will just further cement that this test won't get fixed since it's "good enough".

> We need to ask similiar questions for all cases here.

lol, no one but you understands how these tests work.

Reply:
It won't be easy, for sure. But let's start small, if we can fix one or two then we can see how feasible it is to go further or give up.

> but that will just further cement that this test won't get fixed

If the eq() is not supposed to have a race, yes. But if the state is inherently racy then a permanent retry() is correct.

Reply:
I suspect that something else is using the channel, preventing the channel from being freed.